object_pool.go

```diff
@@ -0,0 +1,45 @@
+package astits
+
+import "sync"
+
+// poolOfTempPayload global variable is used to ease access to pool from any place of the code
+var poolOfTempPayload = &poolTempPayload{
+	sp: sync.Pool{
+		New: func() interface{} {
+			// Prepare the slice of somewhat sensible initial size to minimize calls to runtime.growslice
+			return &tempPayload{
+				s: make([]byte, 0, 1024),
+			}
+		},
+	},
+}
+
+// tempPayload is an object containing payload slice
+type tempPayload struct {
+	s []byte
+}
+
+// poolTempPayload is a pool for temporary payload in parseData()
+// Don't use it anywhere else to avoid pool pollution
+type poolTempPayload struct {
+	sp sync.Pool
+}
+
+// get returns the tempPayload object with byte slice of a 'size' length
+func (ptp *poolTempPayload) get(size int) (payload *tempPayload) {
```

Comment:
Which difference does this function make? Isn't the slice automatically grown when needed?

Reply:
1) We are using this slice to copy existent payload prom packets, so we must be sure that the destination slice length is appropriate 
2) We can replace `copy` with `append`, but if capacity of destination slice is lesser than target length it will lead to grow (potentially multiple grow if pooled slice is to small (e.g. reusing slice for PES after initially using it for PSI)). Also this change will not lift our obligations to reset pooled slice to `[:0]`
3) In this function we growing or trimming the slice to appropriate target length in one operation. It can be seen as drop off replacement for `payload = make([]byte, l)` where we creating slice of known length